src/core_plugins/kibana/public/management/sections/objects/components/objects_table/objects_table.js

```diff
@@ -151,51 +157,66 @@ class ObjectsTableUI extends Component {
     }));
   };
 
-  fetchSavedObjects = async () => {
+  fetchSavedObjects = () => {
+    this.setState({
+      isSearching: true,
+    }, this.debouncedFetch.bind(null, this.state.activeQuery.text));
+  }
+
+  debouncedFetch = debounce(async (activeQueryText) => {
     const { savedObjectsClient } = this.props;
     const { activeQuery, page, perPage } = this.state;
-
-    this.setState({ isSearching: true });
-
     const { queryText, visibleTypes } = parseQuery(activeQuery);
+    const findOptions = {
+      search: queryText ? `${queryText}*` : undefined,
+      perPage,
+      page: page + 1,
+      fields: ['title', 'id'],
+      searchFields: ['title'],
+      type: INCLUDED_TYPES.filter(
+        type => !visibleTypes || visibleTypes.includes(type)
+      ),
+    };
+    if (findOptions.type.length > 1) {
+      findOptions.sortField = 'type';
+    }
 
-    let savedObjects = [];
-    let filteredItemCount = 0;
-
-    const type = INCLUDED_TYPES.filter(
-      type => !visibleTypes || visibleTypes.includes(type)
-    );
-
-    // TODO: is there a good way to stop existing calls if the input changes?
-    await ensureMinimumTime(
-      (async () => {
-        const filteredSavedObjects = await savedObjectsClient.find({
-          search: queryText ? `${queryText}*` : undefined,
-          perPage,
-          page: page + 1,
-          sortField: 'type',
-          fields: ['title', 'id'],
-          searchFields: ['title'],
-          type,
+    let resp;
+    try {
+      resp = await savedObjectsClient.find(findOptions);
+    } catch (error) {
+      if (this._isMounted) {
+        this.setState({
+          isSearching: false,
         });
+      }
+      toastNotifications.addDanger({
+        title: `Unable find saved objects`,
+        text: `${error}`,
+      });
+      return;
+    }
 
-        savedObjects = filteredSavedObjects.savedObjects.map(savedObject => ({
-          title: savedObject.attributes.title,
-          type: savedObject.type,
-          id: savedObject.id,
-          icon: getSavedObjectIcon(savedObject.type),
-        }));
+    if (!this._isMounted) {
+      return;
+    }
 
-        filteredItemCount = filteredSavedObjects.total;
-      })()
-    );
+    // ignore results for old requests that are returned out of order
+    if (activeQueryText !== activeQuery.text) {
```

Comment:
`activeQuery.text` is not updated when the state updates, so in order to do this we need to use a callback for `setState()` and get the current query that way:

```js
this.setState(({ activeQuery: latestActiveQuery }) => {
  if (activeQuery.text !== latestActiveQuery.text) {
    return {};
  }

  return {
    savedObjects: resp.savedObjects.map(savedObject => ({
      title: savedObject.attributes.title,
      type: savedObject.type,
      id: savedObject.id,
      icon: getSavedObjectIcon(savedObject.type),
    })),
    filteredItemCount: resp.total,
    isSearching: false,
  };
});
```

Also, since we're reading `activeQuery` from `this.state` inside `debouncedFetch()` we can probably avoid binding the `activeQueryText` argument, or maybe rename `activeQuery` to just `query` when we destructure `this.state` on line 168